src/include/network/lan/udev.rb

```diff
@@ -82,13 +82,11 @@ def RemoveKeyFromUdevRule(rule, key)
     # Tripplet is given as a string in form KEY="VALUE" or
     # MATCHKEY=="MATCHVALUE"
     def AddToUdevRule(rule, tripplet)
-      rule = deep_copy(rule)
-      if !Builtins.regexpmatch(tripplet, ".+={1,2}\".*\"")
-        return deep_copy(rule)
-      end
+      return rule unless tripplet =~ /.+={1,2}\".*\"/
+
       rule = [] if rule.nil?
 
-      Builtins.add(rule, tripplet)
+      rule << tripplet
```

Comment:
be aware that here you return reference to rule passed as argument, so it slightly change behavior. So just ensure it do not break anything. Problem can be e.g. something like this

```
RULE="c=d"

# old behavior
res = AddToUdevRule(RULE, "a=b")
res = AddToUdevRule(RULE, "a=b")
# res will be "c=d c=d a=b" and RULE will be "c=d"
# in new behavior res and RULE will be "c=d c=d a=b a=b"
```